server/internal/infrastructure/tx-builder/covenantless/builder.go

```diff
@@ -699,30 +699,30 @@ func (b *txBuilder) createPoolTx(
 		return nil, err
 	}
 
-	var dust uint64
-	if change > 0 {
-		if change < dustLimit {
-			dust = change
-			change = 0
-		} else {
-			address, err := b.wallet.DeriveAddresses(ctx, 1)
-			if err != nil {
-				return nil, err
-			}
+	changeAddresses, err := b.wallet.DeriveAddresses(ctx, 1)
+	if err != nil {
+		return nil, err
+	}
 
-			addr, err := btcutil.DecodeAddress(address[0], b.onchainNetwork())
-			if err != nil {
-				return nil, err
-			}
+	changeAddress, err := btcutil.DecodeAddress(changeAddresses[0], b.onchainNetwork())
+	if err != nil {
+		return nil, err
+	}
 
-			aspScript, err := txscript.PayToAddrScript(addr)
-			if err != nil {
-				return nil, err
-			}
+	changeScript, err := txscript.PayToAddrScript(changeAddress)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
not that great to derive a change address in case the change is 0 or below the dust. Please revert